client/src/main/scala/com.crobox.clickhouse/stream/ClickhouseSink.scala

```diff
@@ -49,4 +61,69 @@ object ClickhouseSink extends LazyLogging {
       .mergeSubstreams
       .toMat(Sink.ignore)(Keep.right)
   }
+
+  def toSink(config: Config, client: ClickhouseClient, indexerName: Option[String] = None)(
+      implicit ec: ExecutionContext,
+      settings: QuerySettings = QuerySettings()
+  ): Sink[TableOperation, Future[Done]] = {
+    val indexerGeneralConfig = config.getConfig("crobox.clickhouse.indexer")
+    val mergedIndexerConfig = indexerName
+      .flatMap(
+        theIndexName =>
+          if (indexerGeneralConfig.hasPath(theIndexName))
+            Some(indexerGeneralConfig.getConfig(theIndexName).withFallback(indexerGeneralConfig))
+          else None
+      )
+      .getOrElse(indexerGeneralConfig)
+    val batchSize     = mergedIndexerConfig.getInt("batch-size")
+    val flushInterval = mergedIndexerConfig.getDuration("flush-interval").getSeconds.seconds
+    Flow[TableOperation]
+      .groupBy(Int.MaxValue, _.table)
+      .groupedWithin(batchSize, flushInterval)
+      .mapAsync(mergedIndexerConfig.getInt("concurrent-requests"))(operations => {
+        val table = operations.head.table
+        logger.debug(
+          s"Executing ${operations.size} operations on table: $table. Group Within: ($batchSize - $flushInterval)"
+        )
+
+        insertOp(operations.collect { case o: Insert                     => o }, client)
+          .flatMap(_ => optimizeOp(operations.collect { case o: Optimize => o }, client))
+          .map(_ => operations)
+      })
+      .mergeSubstreams
+      .toMat(Sink.ignore)(Keep.right)
+  }
+
+  private def insertOp(ops: Seq[Insert], client: ClickhouseClient)(
+      implicit ec: ExecutionContext,
+      settings: QuerySettings = QuerySettings()
+  ): Future[Iterable[String]] =
+    Future.sequence(ops.groupBy(_.table).map { t =>
+      val insertQuery = s"INSERT INTO ${t._1} FORMAT JSONEachRow"
+      val payload     = t._2.map(_.jsonRow)
+      logger.info(s"Inserting ${payload.size} entries in table: ${t._1}.")
+      client
+        .execute(insertQuery, payload.mkString("\n"))
+        .recover {
+          case ex => throw ClickhouseIndexingException("failed to index", ex, payload, t._1)
+        }
+    })
+
+  private def optimizeOp(ops: Seq[Optimize], client: ClickhouseClient)(
+      implicit ec: ExecutionContext,
+      settings: QuerySettings = QuerySettings()
+  ): Future[Iterable[String]] = {
+    logger.info(s"Optimizing tables")
+    Future.sequence(
+      ops.map { o =>
+        val query = buildOptimizeQuery(o.table, o.cluster)
```

Comment:
No need to create a seperate def for a single statement. So directly put the  s"OPTIMIZE TABLE $table${cluster.map(s => s"_local ON CLUSTER $s").getOrElse("")} FINAL" here instead of the function call